trieste/acquisition/function.py

```diff
@@ -11,80 +11,84 @@
 
 import tensorflow as tf
 import tensorflow_probability as tfp
+from typing_extensions import final
 
 from ..data import Dataset
-from ..type import QueryPoints
+from ..type import QueryPoints, TensorType
 from ..models import ProbabilisticModel
 from ..space import SearchSpace
 
 from scipy.optimize import bisect
 
+from ..utils import DEFAULTS
 
-AcquisitionFunction = Callable[[QueryPoints], tf.Tensor]
-"""
-Type alias for acquisition functions.
-
+BatchQueryPoints = TensorType
+""" Type alias for batches of `B` query points (each of dimension `D`), with shape `[..., B, D]`."""
 
-An `AcquisitionFunction` maps a single query point (of dimension `D`) to a single value that
-describes how useful it would be evaluate that point (to our goal of optimizing the objective
-function). Thus, with leading dimensions, an `AcquisitionFunction` takes input shape
-`[..., D]` and returns shape `[..., 1]`.
-
-**Note:** Type checkers will not be able to distinguish an `AcquisitionFunction` from a
-`BatchAcquisitionFunction`.
+T = TypeVar("T", QueryPoints, BatchQueryPoints)
 """
+Type variable: either `QueryPoints` or `BatchQueryPoints`.
 
-BatchAcquisitionFunction = Callable[[QueryPoints], tf.Tensor]
-""" 
-Type alias for batch acquisition functions. 
+**Note:** Type checkers will not be able to distinguish `BatchQueryPoints` from `QueryPoints`. They
+are given separate names purely to document intent.
+"""
 
-A `BatchAcquisitionFunction` maps a set of `B` query points (each of dimension `D`) to a single
-value that describes how useful it would be evaluate all these points together (to our goal of
-optimizing the objective function). Thus, with leading dimensions, a `BatchAcquisitionFunction`
-takes input shape `[..., B, D]` and returns shape `[..., 1]`.
+AcquisitionFunction = Callable[[T], tf.Tensor]
+"""
+Type alias for acquisition functions.
 
-**Note:** Type checkers will not be able to distinguish an `AcquisitionFunction` from a
-`BatchAcquisitionFunction`.
+An `AcquisitionFunction` maps query points (each of dimension `D`) to values that describe how
+useful it would be evaluate those points (to our goal of optimizing the objective function).
+
+The argument type can be `QueryPoints` or `BatchQueryPoints`. These work as follows:
+  - If the argument type is `QueryPoints`, the `AcquisitionFunction` evaluates each query point
+    separately. With leading dimensions, this means the function takes input shape `[..., D]` and
+    returns shape `[..., 1]`.
+  - If the argument type is `BatchQueryPoints`, the `AcquisitionFunction` maps each batch of `B`
+    query points to a single value that describes how useful it would be evaluate all these points
+    together. With leading dimensions, this means the function takes input shape `[..., B, D]` and
+    returns shape `[..., 1]`.
 """
 
 
-class AcquisitionFunctionBuilder(ABC):
-    """ An :class:`AcquisitionFunctionBuilder` builds an acquisition function. """
+class AcquisitionBuilder(ABC, Generic[T]):
```

Comment:
why this change? how do we know we're building a function and not a rule? besides, this looks like some work outside this PR

Reply:
simply cos the names were getting long. Can revert it

Reply:
what do you reckon?

Reply:
what do you reckon? We'd then have a `MonteCarloIndependentAcquisitionFunctionBuilder`, and if we add single-model variants, that would be a `SingleModelMonteCarloIndependentAcquisitionFunctionBuilder`. We could use `MC` instead of `MonteCarlo`, or find a shorter way of saying `SingleModel`

Reply:
I'd go for MC - might also use Idpt

Reply:
is Idpt a standard abbrev? cos Ind and Indpt show up on a google search